### PR TITLE
chore: mirgation nextJS

### PR DESCRIPTION
**Pull request by 0xtiti**
## Checklist before requesting a review

- [x] I have conducted a self-review of my code.
- [ ] If it is a core feature, I have included comprehensive tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one sentence about this update.

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Build and Test
+xname: Build and Test
```
Comment (0xArdy) [inline .github/workflows/build-test.yml]: typo?

File: .github/workflows/build-test.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ jobs:
       - name: Install Node
         uses: actions/setup-node@v3
         with:
-          node-version: "18.7"
+          node-version: "18.17"
```
Comment (0xArdy) [inline .github/workflows/build-test.yml]: you can update it to a most recent version `21.4` for example

File: package.json [hunk 3]
```diff
@@ -35,14 +38,14 @@
   "devDependencies": {
     "@defi-wonderland/crypto-husky-checks": "1.3.0",
     "@testing-library/jest-dom": "6.1.2",
-    "@testing-library/react": "14.0.0",
+    "@testing-library/react": "^14.2.1",
```
Comment (0xArdy) [inline package.json]: remove carets, to lock the dependency version

File: package.json [hunk 4]
```diff
@@ -35,14 +38,14 @@
...
     "@testing-library/user-event": "14.4.3",
     "@types/lodash.merge": "4.6.7",
     "@types/react": "18.2.15",
     "@types/react-dom": "18.2.7",
     "@typescript-eslint/eslint-plugin": "6.0.0",
     "@typescript-eslint/parser": "6.0.0",
-    "@vitejs/plugin-react": "4.0.3",
+    "@vitejs/plugin-react": "^4.2.1",
```
Comment (0xArdy) [inline package.json]: you can now remove all vite related dependencies

File: package.json [hunk 5]
```diff
@@ -52,13 +55,11 @@
...
+    "jsdom": "^24.0.0",
     "lodash.merge": "4.6.2",
     "prettier": "3.0.3",
     "styled-components": "6.0.7",
-    "typescript": "5.0.2",
-    "vite": "4.4.5",
-    "vite-tsconfig-paths": "4.2.0",
-    "vitest": "0.34.3"
+    "typescript": "^5.4.2",
+    "vitest": "^1.4.0"
```
Comment (0xArdy) [inline package.json]: idem.

Please try to remove all dependencies that aren't used.

File: next.config.mjs [hunk 6]
```diff
@@ -0,0 +1,7 @@
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  output: 'export', // Outputs a Single-Page Application (SPA).
+  distDir: './dist', // Changes the build output directory to `./dist/`.
```
Comment (0xArdy) [inline next.config.mjs]: I would prefer to leave the default settings

File: src/containers/Footer/index.tsx [hunk 7]
```diff
@@ -1,40 +1,9 @@
-import styled from 'styled-components';
-
 export const Footer = () => {
   return (
-    <FooterContainer>
+    <>
```
Comment (0xArdy) [inline src/containers/Footer/index.tsx]: use a `footer` tag instead of a react fragment

File: src/containers/Header/index.tsx [hunk 8]
```diff
@@ -18,31 +16,10 @@ export const Header = () => {
   };
 
   return (
-    <HeaderContainer>
+    <div>
```
Comment (0xArdy) [inline src/containers/Header/index.tsx]: use a `header` tag as a wrapper

File: src/containers/Layout/index.tsx [hunk 9]
```diff
@@ -1,13 +1,16 @@
-import { Outlet } from 'react-router-dom';
-
+import { ReactNode } from 'react';
 import { Footer } from '~/containers/Footer';
 import { Header } from '~/containers/Header';
```
Comment (0xArdy) [inline src/containers/Layout/index.tsx]: I think you can group those imports
```suggestion
import { Header, Footer } from '~/containers';
```

File: vitest.config.ts [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+  plugins: [react()],
+  test: {
+    environment: 'jsdom',
+  },
+  resolve: {
+    alias: {
+      '~/': `${resolve(__dirname, 'src')}/`,
+    },
+  },
+});
```
Comment (0xArdy) [inline vitest.config.ts]: i think this dependency and this file should be removed. `vitest` should be replaced with `jest` or `cypress` what do you think?

File: .eslintrc.cjs [hunk 11]
```diff
@@ -2,22 +2,33 @@ module.exports = {
...
-      { allowConstantExport: true },
-    ],
+    'react-refresh/only-export-components': ['warn', { allowConstantExport: true }],
+    // Additional rules can be added here
   },
+  overrides: [
+    {
+      files: ['*.ts', '*.tsx'],
+      rules: {
+        'react/prop-types': 'off', // Turn off prop-types rule for TypeScript files
```
Comment (0xArdy) [inline .eslintrc.cjs]: why disable it?

File: src/pages/_app.tsx [hunk 12]
```diff
@@ -1,19 +1,16 @@
-import React from 'react';
-import ReactDOM from 'react-dom/client';
-import { HashRouter } from 'react-router-dom';
-
+import { AppProps } from 'next/app';
+import { Themable } from '~/components';
+import { ScrollToTop } from '~/hooks';
+import { StateProvider } from '~/providers';
```
Comment (0xArdy) [inline src/pages/_app.tsx]: library imports should be above our imported files

**Review by 0xArdy — CHANGES_REQUESTED** [review]
great work!

File: package.json (hunk 3 above)
Comment (0xtiti) [reply package.json]: https://github.com/defi-wonderland/web3-react-boilerplate/commit/af82dd526ba9023b2a88ca17bbe35c75120fa320

File: src/pages/_app.tsx (hunk 12 above)
Comment (0xtiti) [reply src/pages/_app.tsx]: https://github.com/defi-wonderland/web3-react-boilerplate/commit/9b5c51f0ad472ae9a2584c0f0d364b6e6f83e685

File: src/containers/Layout/index.tsx (hunk 9 above)
Comment (0xtiti) [reply src/containers/Layout/index.tsx]: https://github.com/defi-wonderland/web3-react-boilerplate/commit/9b5c51f0ad472ae9a2584c0f0d364b6e6f83e685

File: src/containers/Header/index.tsx (hunk 8 above)
Comment (0xtiti) [reply src/containers/Header/index.tsx]: https://github.com/defi-wonderland/web3-react-boilerplate/commit/9b5c51f0ad472ae9a2584c0f0d364b6e6f83e685

File: src/containers/Footer/index.tsx (hunk 7 above)
Comment (0xtiti) [reply src/containers/Footer/index.tsx]: https://github.com/defi-wonderland/web3-react-boilerplate/commit/9b5c51f0ad472ae9a2584c0f0d364b6e6f83e685

File: next.config.mjs (hunk 6 above)
Comment (0xtiti) [reply next.config.mjs]: https://github.com/defi-wonderland/web3-react-boilerplate/commit/a77794c01eb09d00167ff7b2882557664a9d21ec

File: .eslintrc.cjs (hunk 11 above)
Comment (0xtiti) [reply .eslintrc.cjs]: https://github.com/defi-wonderland/web3-react-boilerplate/commit/1b2facf5b8b62565ab21d1b3295d08feac6c0cd2

File: package.json (hunk 4 above)
Comment (0xtiti) [reply package.json]: https://github.com/defi-wonderland/web3-react-boilerplate/commit/9a9683fc0ce55ee0a3c687f7ac80783cade96b46

File: package.json (hunk 5 above)
Comment (0xtiti) [reply package.json]: https://github.com/defi-wonderland/web3-react-boilerplate/commit/9a9683fc0ce55ee0a3c687f7ac80783cade96b46

File: vitest.config.ts (hunk 10 above)
Comment (0xtiti) [reply vitest.config.ts]: https://github.com/defi-wonderland/web3-react-boilerplate/commit/9a9683fc0ce55ee0a3c687f7ac80783cade96b46

File: package.json [hunk 13]
```diff
@@ -35,30 +40,28 @@
...
-    "jsdom": "22.1.0",
+    "jsdom": "24.0.0",
     "lodash.merge": "4.6.2",
     "prettier": "3.0.3",
     "styled-components": "6.0.7",
-    "typescript": "5.0.2",
-    "vite": "4.4.5",
-    "vite-tsconfig-paths": "4.2.0",
-    "vitest": "0.34.3"
+    "typescript": "^5.4.2"
```
Comment (0xArdy) [inline package.json]: remove carets

**Review by 0xArdy — CHANGES_REQUESTED** [review]
just one smol comment

File: package.json (hunk 13 above)
Comment (0xtiti) [reply package.json]: https://github.com/defi-wonderland/web3-react-boilerplate/commit/58f0ce8e298eb1981585506f7ce2a02a0923c6ed

**Review by 0xArdy — APPROVED** [review]

**Pull request merged**